cubano-httpeasy/src/main/java/org/concordion/cubano/driver/http/HttpEasy.java

```diff
@@ -1,880 +1,869 @@
+ * <p>
+ * * request().doNotFailOn(Integer... reponseCodes)
+ * * request().doNotFailOn(Family... responseFamily)
+ * </p>
+ * <p>
+ * <b>Authentication</b>
+ * </p>
+ * <p>
+ * Supports two formats
+ * <p>
+ * * http://username:password@where.ever
+ * * request().authorization(username, password)
+ * </p>
+ * <p>
+ * <b>Host and Certificate Verification</b>
+ * </p>
+ * <p>
+ * There is no fine grained control, its more of an all or nothing approach:
+ * </p>
+ * <pre>
+ * HttpEasy.withDefaults()
+ *      .allowAllHosts()
+ *      .trustAllCertificates();
+ * </pre>
+ * <p>
+ * <b>Proxy</b>
+ * </p>
+ * <p>
+ * Only basic authentication is supported, although I believe the domain can be added by included "domain/"
+ * in front of the username (not tested)
+ * </p>
+ * <pre>
+ * HttpEasy.withDefaults()
+ *     .proxy(new Proxy(Proxy.Type.HTTP, new InetSocketAddress(user, password))))
+ *     .proxyAuth(userName, password)
+ *     .bypassProxyForLocalAddresses(true);
+ * </pre>
+ * <p>
+ * <b>Redirects</b>
+ * </p>
+ * <p>
+ * Redirects are NOT automatically followed - at least for REST base calls - even though the documentation
+ * for HttpURLConnection says that it should...
+ * </p>
+ * <pre>
+ * HttpEasyReader response = HttpEasy.request()
+ *     .doNotFailOn(Family.REDIRECTION)
+ *     .path(url)
+ *     .head();
+ *
+ * if (response.getResponseCodeFamily() == Family.REDIRECTION) {
+ *     url = response.getHeaderField("Location");
+ *     ...
+ * }
+ * </pre>
+ * <p>
+ * <b>Logging</b>
+ * </p>
+ * <p>
+ * Logging of requests and responses can be enabled by {@link #logRequestDetails} or, if using Eclipse, the TCP/IP Monitor utility.
+ * </p>
+ */
+public class HttpEasy {
+    static final Logger LOGGER = LoggerFactory.getLogger(HttpEasy.class);
+
+    // These only apply per request - but are visible to package
+    List<Integer> ignoreResponseCodes = new ArrayList<Integer>();
+    List<Family> ignoreResponseFamily = new ArrayList<Family>();
+
+    // These only apply per request
+    private String authString = null;
+    private String baseURI = "";
+    private String path = "";
+    private StringBuilder query = new StringBuilder();
+    private String startToken = "{";
+    private String endToken = "}";
+    private Object[] urlParams = new Object[0];
+    private DataContentType dataContentType = DataContentType.AUTO_SELECT;
+    private Object rawData = null;
+    private String rawFileName = null;
+    private MediaType rawDataMediaType = null;
+    private Map<String, Object> headers = new LinkedHashMap<String, Object>();
+    private List<Field> fields = new ArrayList<Field>();
+    private LogWriter logWriter = null;
+    private boolean logRequestDetails;
+    private Integer timeout = null;
+
+    boolean isLogRequestDetails() {
+        return logRequestDetails;
+    }
+
+    static {
+        // TODO Ideally httpeasy wouldn't dependencies on other components to make it easier to use outside of cubano

```

Comment:
Why not consider dependency inversion, eg create a parameterised method taking host, port, user and password, then pass those values in from cubano-core?

Reply:
Like the idea, I chat to you about where best to do this next time we catch up